attestation-service/verifier/src/az_tdx_vtpm/mod.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) Microsoft Corporation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use super::tdx::claims::generate_parsed_claim;
+use super::tdx::quote::{ecdsa_quote_verification, parse_tdx_quote, Quote as TdQuote};
+use super::{TeeEvidenceParsedClaim, Verifier};
+use crate::{regularize_data, InitDataHash, ReportData};
+use anyhow::{bail, Context, Result};
+use async_trait::async_trait;
+use az_tdx_vtpm::hcl::HclReport;
+use az_tdx_vtpm::vtpm::Quote as TpmQuote;
+use log::{debug, warn};
+use openssl::pkey::PKey;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+struct Evidence {
+    tpm_quote: TpmQuote,
+    hcl_report: Vec<u8>,
+    td_quote: Vec<u8>,
+}
+
+#[derive(Default)]
+pub struct AzTdxVtpm;
+
+#[async_trait]
+impl Verifier for AzTdxVtpm {
+    /// The following verification steps are performed:
+    /// 1. TPM Quote has been signed by AK included in the HCL variable data
+    /// 2. Attestation nonce matches TPM Quote nonce
+    /// 3. TD Quote is genuine
+    /// 4. TD Report's report_data field matches hashed HCL variable data
+    async fn evaluate(
+        &self,
+        evidence: &[u8],
+        expected_report_data: &ReportData,
+        expected_init_data_hash: &InitDataHash,
+    ) -> Result<TeeEvidenceParsedClaim> {
+        let ReportData::Value(expected_report_data) = expected_report_data else {
+            bail!("unexpected empty report data");
+        };
+        let expected_report_data =
+            regularize_data(expected_report_data, 64, "REPORT_DATA", "Azure TDX vTPM");
+
+        if let InitDataHash::Value(_) = expected_init_data_hash {
+            warn!("Azure TDX vTPM verifier does not support verify init data hash, will ignore the input `init_data_hash`");
+        }
+
+        let evidence = serde_json::from_slice::<Evidence>(evidence)
+            .context("Failed to deserialize Azure vTPM TDX evidence")?;
+
+        let hcl_report = HclReport::new(evidence.hcl_report)?;
+        verify_tpm_quote(&evidence.tpm_quote, &hcl_report, &expected_report_data)?;
+
+        ecdsa_quote_verification(&evidence.td_quote).await?;
+        let td_quote = parse_tdx_quote(&evidence.td_quote)?;
+
+        verify_report_data(&hcl_report, &td_quote)?;
+
+        let claim = generate_parsed_claim(td_quote, None)?;
```

Comment:
I think the TDX claims parsing has been updated to include an i[nit_data field](https://github.com/confidential-containers/kbs/blob/5210b8651bff63abf5294e01ca0f30744bc3714e/attestation-service/verifier/src/tdx/claims.rs#L128).

Is this something that we want to include for the AZ case or should we remove that key like we were talking about for SNP.

Reply:
Ok, you're saying that since this attester will (only?) be used with peer pods we can point the init_data field to a PCR, which will have been used to measure the init_data?

Reply:
yes

Reply:
(independent of peerpods, a user wouldn't be able to seed init_data into HW reports when using those specific Azure TEE's. Dynamic aspects like report_data and init_data would have to be covered via TPM)

Reply:
Yeah at first I was a little concerned that this wouldn't be generic enough and would be weird for people not using peer pods, but I guess nobody will be able to use the init data anyway. So you might as well give them init data from some PCR and hope that people find the documentation for it.

Reply:
I wouldn't know another way to implement measured config-data - at least for the type that is used to bootstrap confidentiality, so things like aa_kbc_params, kbs cert.